apps/heureka/src/components/Services/utils.ts

```diff
@@ -39,35 +40,60 @@ type NormalizedServices = {
   services: ServiceType[]
 }
 
-export const getNormalizedData = (data: GetServicesQuery | undefined): NormalizedServices => ({
-  totalCount: data?.Services?.totalCount || 0,
-  pages: data?.Services?.pageInfo?.pages?.filter((edge) => edge !== null) || [],
-  services: isNil(data?.Services?.edges)
-    ? []
-    : data?.Services?.edges
-        ?.filter((edge) => edge !== null)
-        .map((edge?: Edge): ServiceType => {
-          const node: Service | undefined = edge?.node
-          const service: ServiceType = {
-            id: node?.id?.toString() || "",
-            name: node?.ccrn || "",
-            serviceDetails: {
-              supportGroups: getSupportGroups(edge),
-            },
-            components: node?.objectMetadata?.componentInstanceCount || 0,
-            serviceOwners: getServiceOwners(edge),
-            issuesCount: {
-              critical: Math.floor(Math.random() * 10), //TODO: remove mock data when available
-              high: Math.floor(Math.random() * 10), //TODO: remove mock data when available
-            },
-            remediationDate: "2023-01-01", //TODO: remove mock data when available
-          }
-          return service
-        }),
-})
+type ExtendedService = Service & {
+  critical?: IssueMatchConnection
+  high?: IssueMatchConnection
+  medium?: IssueMatchConnection
+  low?: IssueMatchConnection
+  none?: IssueMatchConnection
+}
 
-export const getNormalizedError = (error?: ApolloError) =>
-  !isNil(error) ? error?.message || "Something went wrong" : undefined
+export const getNormalizedData = (data: GetServicesQuery | undefined): NormalizedServices => {
+  return {
+    totalCount: data?.Services?.totalCount || 0,
+    pages: data?.Services?.pageInfo?.pages?.filter((edge) => edge !== null) || [],
+    services: isNil(data?.Services?.edges)
+      ? []
+      : data?.Services?.edges
+          ?.filter((edge) => edge !== null)
+          .map((edge?: Edge): ServiceType => {
+            const node: ExtendedService | undefined = edge?.node
+            const service: ServiceType = {
+              id: node?.id?.toString() || "",
+              name: node?.ccrn || "",
+              serviceDetails: {
+                supportGroups: getSupportGroups(edge),
+              },
+              components: node?.objectMetadata?.componentInstanceCount || 0,
+              serviceOwners: getServiceOwners(edge),
+              issuesCount: {
+                critical: node?.critical?.totalCount || 0,
+                high: node?.high?.totalCount || 0,
+              },
+              remediationDate: "2023-01-01", //TODO: remove mock data when available
+            }
+            return service
+          }),
+  }
+}
+
+export const getNormalizedError = (error?: ApolloError) => {
+  if (isNil(error)) return undefined
+
+  // Extract network errors if they exist
+  const networkErrors = error.networkError as any
```

Comment:
Apollo has a type for it called `NetworkError` I think we could use that instead `any`. But a general question, do we really need to distinguish between network and graphql error at the moment? Because, if so, then I'll suggest doing it properly by configuring an error link like mentioned [here](https://www.apollographql.com/docs/react/data/error-handling#advanced-error-handling-with-apollo-link)

Reply:
yes you are right!

Reply:
we keep it for now as any since we have an extra task for it